File: scripts/gc_translation/paths.py
"""Central path constants for the GC translation pipeline."""
from pathlib import Path

REPO_ROOT = Path(__file__).resolve().parents[2]

DATA_DIR = REPO_ROOT / "data"
SOURCE_DIR = DATA_DIR / "gc-source"
CHUNK_DIR = SOURCE_DIR / "chunks"
TRANSLATED_DIR = DATA_DIR / "gc-translated"

GLOSSARY_PATH = DATA_DIR / "gc-translation" / "glossary.yaml"
BIBLE_REFS_PATH = DATA_DIR / "gc-translation" / "bible-refs.yaml"
VI1934_PATH = DATA_DIR / "bible" / "vi1934.json"

HUGO_BOOK_DIR = REPO_ROOT / "content" / "sach" / "egw" / "thien-ac-dau-tranh"

CHAPTERS = 42
BOOK_ID = 132  # egwwritings book id


def chapter_source_html(n: int) -> Path:
    return SOURCE_DIR / f"ch{n:02d}.html"


def chapter_source_text(n: int) -> Path:
    return SOURCE_DIR / f"ch{n:02d}.txt"


def chunk_path(chapter: int, chunk: int) -> Path:
    return CHUNK_DIR / f"ch{chapter:02d}-{chunk:02d}.txt"


def translated_path(chapter: int, chunk: int) -> Path:
    return TRANSLATED_DIR / f"ch{chapter:02d}-{chunk:02d}.md"


def error_path(chapter: int, chunk: int) -> Path:
    return TRANSLATED_DIR / f"ch{chapter:02d}-{chunk:02d}.err"


def hugo_chapter_path(n: int) -> Path:
    return HUGO_BOOK_DIR / f"chuong-{n:02d}.md"

File: scripts/egw_translate/lib/paths.ts
import { join } from "path";
import type { TranslateBookConfig } from "./types";

const REPO_ROOT = `${import.meta.dir}/../../..`;

export function absPath(book: TranslateBookConfig, relative: string): string {
  return join(REPO_ROOT, relative);
}

export function sourceTextPath(book: TranslateBookConfig, chapter: number): string {
  const nn = String(chapter).padStart(2, "0");
  return join(REPO_ROOT, book.sourceDir, `ch${nn}.txt`);
}

export function chunkPath(book: TranslateBookConfig, chapter: number, chunk: number): string {
  const nn = String(chapter).padStart(2, "0");
  const mm = String(chunk).padStart(2, "0");
  return join(REPO_ROOT, book.chunksDir, `ch${nn}-${mm}.txt`);
}

export function translatedChunkPath(book: TranslateBookConfig, chapter: number, chunk: number): string {
  const nn = String(chapter).padStart(2, "0");
  const mm = String(chunk).padStart(2, "0");
  return join(REPO_ROOT, book.translatedDir, `ch${nn}-${mm}.md`);
}

export function errorChunkPath(book: TranslateBookConfig, chapter: number, chunk: number): string {
  const nn = String(chapter).padStart(2, "0");
  const mm = String(chunk).padStart(2, "0");
  return join(REPO_ROOT, book.translatedDir, `ch${nn}-${mm}.err`);
}

export function hugoChapterPath(book: TranslateBookConfig, chapter: number): string {
  const nn = String(chapter).padStart(2, "0");
  return join(REPO_ROOT, book.hugoBookDir, `chuong-${nn}.md`);
}

export function chaptersJsonPath(book: TranslateBookConfig): string {
  return join(REPO_ROOT, book.sourceDir, "chapters.json");
}
